buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy

```diff
@@ -204,7 +204,11 @@ class PluginBuildPlugin implements Plugin<Project> {
         // create the actual bundle task, which zips up all the files for the plugin
         TaskProvider<Zip> bundle = project.tasks.register('bundlePlugin', Zip) {
             from buildProperties
-            from pluginMetadata // metadata (eg custom security policy)
+            from(pluginMetadata) {
```

Comment:
Rather than do this why don't we just define `pluginMetadata` up top as the individual file isntead of the entire `plugin-metadata` folder and then explicitly include only a single file?

Also, is there any use case for throwing other stuff in that location? Could external plugin authors being doing this, therefore this be a breaking change? Would it be better to explicitly *exclude* the codebases file?

Reply:
We need the pluginMetadata above to still contain everything, which is the policy file and the codebases file, because that is what goes into the test resources dir. We really just want to remove the file when building the jar, which is what we do here.

In theory external plugin authors could add other stuff to the plugin metadata. I could flip this so that it is an exclude of the codebases file. However, there are other ways plugin authors could add files to the root of their plugin zip, so I think it's ok this convention is our own, and not general purpose.

Reply:
Right, I think you misread the first part of my comment though. If indeed we just want to add _only_ the `plugin-security.policy` file to the jar, why not simplify it to `File pluginMetadata = project.file('src/main/plugin-metadata/plugin-security.policy')` and call it a day rather than say "copy this directory, but filter everything but this one file".

Reply:
That is fair. I'd rather make that decision whether we should only copy the policy file separately, so for now I've flipped the logic to exclude the codebases file.